README.md

```diff
@@ -223,895 +103,270 @@ const config = [
-const projector = new AppProjector();
-
-projector.append();
-router.start();
+]
 ```
 
-```ts
-// Home.ts
-import { WidgetBase } from '@dojo/widget-core/WidgetBase';
-import { DNode } from '@dojo/widget-core/interfaces';
+### Router
 
-export class Home extends WidgetBase {
-	protected render(): DNode {
-		return 'Home';
-	}
-}
-```
+A `Router` registers a [route configuration](#route-configuration) which is passed to the router on construction:
 
 ```ts
-import { WidgetBase } from '@dojo/widget-core/WidgetBase';
-import { w, v } from '@dojo/widget-core/d';
-import { DNode, WidgetProperties } from '@dojo/widget-core/interfaces';
-import { Link } from '@dojo/routing/Link';
-
-import { ProfileOutlet } from './ProfileOutlet';
-
-export interface ProfilesProperties extends WidgetProperties {
-	showHeader: boolean;
-}
-
-export class Profiles extends WidgetBase<ProfilesProperties> {
-	protected render(): DNode {
-		return [
-			v('div', [
-				w(Link, { to: 'profile', params: { profile: 'Tess' } }, [ 'Tess ']),
-				w(Link, { to: 'profile', params: { profile: 'Jess' } }, [ 'Jess ']),
-				w(Link, { to: 'profile', params: { profile: 'Bess' } }, [ 'Bess '])
-			]),
-			v('div', [
-				this.properties.showHeader ? 'Please select a profile' : null,
-				w(ProfileOutlet, {})
-			])
-		];
-	}
-}
+const router = new Router(config);
 ```
 
-```ts
-// Profile.ts
-import { WidgetBase } from '@dojo/widget-core/WidgetBase';
-import { DNode, WidgetProperties } from '@dojo/widget-core/interfaces';
-
-export interface ProfileProperties extends WidgetProperties {
-	name: string;
-}
-
-export class Profile extends WidgetBase<ProfileProperties> {
-	protected render(): DNode {
-		return `Hello, ${this.properties.name}`;
-	}
-}
-```
+The router will automatically be registered with a `HashHistory` history manager. This can be overridden by passing a different history manager as the second parameter.
 
 ```ts
-// HomeOutlet.ts
-import { Outlet } from '@dojo/routing/Outlet';
-
-import { Home } from './Home';
+import { MemoryHistory } from '@dojo/routing/MemoryHistory';
 
-export const HomeOutlet = Outlet(Home, 'home');
+const router = new Router(config, MemoryHistory);
 ```
 
-```ts
-// ProfilesOutlet.ts
-import { Outlet } from '@dojo/routing/Outlet';
-import { MatchType } from '@dojo/routing/Route';
-
-import { Profiles } from './Profiles';
-
-export const ProfilesOutlet = Outlet(Profiles, 'profiles', ({ type }: MapParamsOptions) => {
-	return { showHeader: type === MatchType.INDEX };
-});
-```
+Once the router has been created with the application route configuration it needs to be made available to all the components within your application. This is done using a `Registry` from `@dojo/widget-core/Registry` and defining an `Injector` that contains the `router` instance as the `payload`. This `Injector` is defined using a known key, by default through `@dojo/routing` the key is `router` but can be overridden throughout if desired.
```

Comment:
"This `Injector` is defined using a known key, by default through `@dojo/routing` the key is `router` but can be overridden throughout if desired" is kind of run-on-y